nvi-commons/src/test/java/no/sikt/nva/nvi/common/service/NviGraphValidatorTest.java

```diff
@@ -0,0 +1,1416 @@
+    };
+  }
+
+  private static String createObjectFromString(String string) {
+    return string.startsWith("Class:")
+        ? ":" + string.replaceAll("Class:", "")
+        : "\"" + string + "\"";
+  }
+
+  private Model removePublicationDateYear(Model model) {
+    return removeTriples(model, removeQuery(PUBLICATION_DATE_CLASS, YEAR_PROPERTY));
+  }
+
+  private Model addPublicationDateYear(Model model) {
+    return addTriples(model, addQuery(PUBLICATION_DATE_CLASS, YEAR_PROPERTY, "7777"));
+  }
+
+  @Test
+  void shouldReportWhenPublicationDateTypeIsUndefined() {
+    var model = createModelWithNoErrors();
+    var validation = nviGraphValidator.validate(removePublicationDateType(model));
+    assertThat(validation.generateReport())
+        .containsSequence("Publication date type is undefined")
+        .hasSize(1);
+  }
+
+  private Model removePublicationDateType(Model model) {
+    var query =
+        """
+        PREFIX : <%s>
+        CONSTRUCT {
+          ?subject a :PublicationDate .
+        } WHERE {
+          [] :publicationDate ?subject .
+          ?subject a :PublicationDate .
+        }
+        """
+            .formatted(NVA_ONTOLOGY);
+    return removeTriples(model, query);
+  }
+
+  private Model removePublicationChannelType(Model model) {
+    return removeTriples(model, removeQuery(PUBLICATION_CHANNEL_CLASS, "channelType"));
+  }
+
+  private Model addPublicationChannelType(Model model) {
+    var query =
+        """
+        PREFIX : <%s>
+        CONSTRUCT {
+          ?subject :channelType ?channelType .
+        } WHERE {
+          ?subject a :PublicationChannel ;
+                   :channelType ?current .
+          BIND(IF(?current = :Journal, :Publisher, :Journal) AS ?channelType)
+        }
+        """
+            .formatted(NVA_ONTOLOGY);
+    return addTriples(model, query);
+  }
+
+  private Model replacePublicationChannelType(Model model) {
+    var removeModel = removeTriples(model, removeQuery(PUBLICATION_CHANNEL_CLASS, "channelType"));
+    return addTriples(
+        removeModel,
+        addQuery(PUBLICATION_CHANNEL_CLASS, "channelType", "Class:NonsensicalChannelType"));
+  }
+
+  private Model removePublicationChannelName(Model model) {
+    return removeTriples(model, removeQuery(PUBLICATION_CHANNEL_CLASS, NAME_PROPERTY));
+  }
+
+  private Model addPublicationChannelName(Model model) {
+    return addTriples(
+        model, addQuery(PUBLICATION_CHANNEL_CLASS, NAME_PROPERTY, "Another name for the thing"));
+  }
+
+  private Model replacePublicationChannelName(Model model) {
+    var removeModel = removeTriples(model, removeQuery(PUBLICATION_CHANNEL_CLASS, NAME_PROPERTY));
+    return addTriples(
+        removeModel,
+        addQuery(PUBLICATION_CHANNEL_CLASS, NAME_PROPERTY, URI.create("https://example.org/abc")));
+  }
+
+  private Model removePublicationChannelIdentifier(Model model) {
+    return removeTriples(model, removeQuery(PUBLICATION_CHANNEL_CLASS, IDENTIFIER_PROPERTY));
+  }
+
```

Comment:
Also check for "scientificValue" : "Unassigned" maybe?

Reply:
See update